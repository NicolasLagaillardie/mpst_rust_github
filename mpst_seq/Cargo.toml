[package]
name = "mpst-seq"
version = "0.0.7"
authors = ["NicolasLagaillardie <lagaillardie.nicolas@live.fr>"]
description = "A library with macro for mpstthree"
edition = "2018"
readme = "README.md"
repository = "https://github.com/NicolasLagaillardie/mpst_rust_github/tree/master/mpst_seq/"
license = "MIT"
exclude = ["mpst_seq_proc/"]
autotests = false

[workspace]
exclude = ["mpst_seq_proc/"]

[dev-dependencies]
trybuild = "1.0.42"

[dependencies]
proc-macro-hack = "0.5.19"
mpst-seq-proc = { path = "mpst_seq_proc", version = "0.0.7" }
# mpst-seq-proc = "0.0.7"
