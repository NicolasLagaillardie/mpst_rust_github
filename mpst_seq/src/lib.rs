extern crate proc_macro;

use proc_macro::TokenStream;

#[proc_macro]
pub fn seq(input: TokenStream) -> TokenStream {
    let _ = input;

    unimplemented!()
}
