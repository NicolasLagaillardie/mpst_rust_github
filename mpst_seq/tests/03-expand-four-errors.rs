// Now construct the generated code! Produce the output TokenStream by repeating
// the loop body the correct number of times as specified by the loop bounds and
// replacing the specified identifier with the loop counter.
//
// The invocation below will need to expand to a TokenStream containing:
//
//     compile_error!(concat!("error number ", stringify!(0)));
//     compile_error!(concat!("error number ", stringify!(1)));
//     compile_error!(concat!("error number ", stringify!(2)));
//     compile_error!(concat!("error number ", stringify!(3)));
//
// This test is written as a compile_fail test because our macro isn't yet
// powerful enough to do anything useful. For example if we made it generate
// something like a function, every one of those functions would have the same
// name and the program would not compile.

use mpst_seq::seq;

seq!(N in 0..4 {
    compile_error!(concat!("error number ", stringify!(N)));
});

fn main() {}
