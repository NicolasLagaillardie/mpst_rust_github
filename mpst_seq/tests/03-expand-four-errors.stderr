error: error number 0
  --> $DIR/03-expand-four-errors.rs:20:5
   |
20 |     compile_error!(concat!("error number ", stringify!(N)));
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error: error number 1
  --> $DIR/03-expand-four-errors.rs:20:5
   |
20 |     compile_error!(concat!("error number ", stringify!(N)));
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error: error number 2
  --> $DIR/03-expand-four-errors.rs:20:5
   |
20 |     compile_error!(concat!("error number ", stringify!(N)));
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error: error number 3
  --> $DIR/03-expand-four-errors.rs:20:5
   |
20 |     compile_error!(concat!("error number ", stringify!(N)));
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
